src/api/shopping.py

```diff
@@ -78,49 +63,38 @@ def optimize_shopping_route(user_id: int, food_id: int, use_preferences: bool):
                     detail=f"Food item does not exist, {e}"
                 )
 
-            valid_stores = [
-                {
-                    "name": row.store_name,
-                    "store_id": row.store_id,
-                    "price": row.price,
-                    "distance": geodesic((user_info.latitude, user_info.longitude), (row.latitude, row.longitude)).km
-                }
-                for row in result
-            ]
+    valid_stores = [
+        {
+            "name": row.store_name,
+            "store_id": row.store_id,
+            "price": row.price,
+            "distance": geodesic((user_info.latitude, user_info.longitude), (row.latitude, row.longitude)).km
+        }
+        for row in result
+    ]
 
     if not valid_stores:
         raise HTTPException(
             status_code=status.HTTP_404_NOT_FOUND,
             detail="No stores found with the requested item"
         )
 
-    if not use_preferences:
-        # Sort by distance and get closest store
-        valid_stores.sort(key=lambda store: store["distance"])
-        closest_store = valid_stores[0]
-        return {
-            "Closest Store": {
-                "Name": closest_store["name"],
-                "Store ID": closest_store["store_id"],
-                "Distance Away": f"{closest_store['distance']:.2f} km",
-                "Price of Item": f"${closest_store['price'] / 100:.2f}"
-            }
-        }
-    else:
+    if budget > 0:
```

Comment:
Much prefer to see this filtering done in the sql statement. I've done it before where you have something like:
```
sql_string = "SELECT * FROM table WHERE id = :id"
if budget > 0:
    sql_string += " AND price < :budget"
```
You can just have an if statement that tacks on the extra where clause.